bundles/org.openhab.core.io.rest.auth/src/main/java/org/openhab/core/io/rest/auth/internal/AuthFilter.java

```diff
@@ -207,47 +200,62 @@ private SecurityContext authenticateBasicAuth(String credentialString) throws Au
                 String altTokenHeader = requestContext.getHeaderString(ALT_AUTH_HEADER);
                 if (altTokenHeader != null) {
-                    requestContext.setSecurityContext(authenticateBearerToken(altTokenHeader));
+                    cacheKey = getCacheKey(altTokenHeader);
+                    SecurityContext securityContext = authCache.get(cacheKey);
+                    if (securityContext == null) {
+                        securityContext = authenticateBearerToken(altTokenHeader);
+                        authCache.put(cacheKey, securityContext, null);
+                    }
+                    requestContext.setSecurityContext(securityContext);
                     return;
                 }
 
                 String authHeader = requestContext.getHeaderString(HttpHeaders.AUTHORIZATION);
                 if (authHeader != null) {
+                    cacheKey = getCacheKey(authHeader);
+                    SecurityContext securityContext = authCache.get(cacheKey);
+                    if (securityContext != null) {
+                        requestContext.setSecurityContext(securityContext);
+                        return;
+                    }
+
                     String[] authParts = authHeader.split(" ");
                     if (authParts.length == 2) {
                         String authType = authParts[0];
                         String authValue = authParts[1];
 
                         if ("Bearer".equalsIgnoreCase(authType)) {
-                            requestContext.setSecurityContext(authenticateBearerToken(authValue));
+                            securityContext = authenticateBearerToken(authValue);
+                            authCache.put(cacheKey, securityContext, null);
+                            requestContext.setSecurityContext(securityContext);
                             return;
                         } else if ("Basic".equalsIgnoreCase(authType)) {
                             if (!allowBasicAuth) {
                                 throw new AuthenticationException(
                                         "Basic authentication with username/password is not allowed");
                             }
-                            requestContext.setSecurityContext(authenticateBasicAuth(authValue));
+                            securityContext = authenticateBasicAuth(authValue);
+                            authCache.put(cacheKey, securityContext, null);
+                            requestContext.setSecurityContext(securityContext);
                         }
                     }
                 } else if (implicitUserRole) {
                     requestContext.setSecurityContext(new AnonymousUserSecurityContext());
                 }
             } catch (AuthenticationException e) {
-                logger.warn("Unauthorized API request: {}", e.getMessage());
+                String errorMessage = e.getMessage();
+                authCache.put(cacheKey, null, errorMessage);
```

Comment:
Why is this useful?

Reply:
If unauthorized requests are done, it is imho good if they do not return quickly.
For any DoS-style attack, we should rather think about putting some throttling in place, but not to increase the response speed, which is rather counter-productive.
If a "friendly" app keeps firing requests even though it sees that they are unauthenticated, the logic should rather be changed within the app.